lms/validation/_api.py

```diff
@@ -132,3 +137,31 @@ def validate(self, data, **_kwargs):
             raise ValidationError(
                 "both bookID and cfi are mandatory with type is 'vitalsource'"
             )
+
+
+class APISyncBlackboardSchema(PyramidRequestSchema):
+    class Data(Schema):
+        class LMS(Schema):
+            tool_consumer_instance_guid = fields.Str(required=True)
+
+        class Course(Schema):
+            context_id = fields.Str(required=True)
+
+        class Assignment(Schema):
+            resource_link_id = fields.Str(required=True)
+
+        class GroupInfo(Schema):
+            class Meta:
+                unknown = INCLUDE
+
+        lms = fields.Nested(LMS, required=True)
+        course = fields.Nested(Course, required=True)
+        assignment = fields.Nested(Assignment, required=True)
+        group_info = fields.Nested(GroupInfo, required=True)
+
+        gradingStudentId = fields.Str(required=False, allow_none=True)
```

Comment:
This one is not sent by use, it's injected by the frontend so the naming has JS conventions.